adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,32 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Converts this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage(collapse: Boolean = true): CoverageRDD = {
+    val coverage: RDD[Coverage] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+        }).reduceByKey(_ + _)
+        .cache()
```

Comment:
Also, why are we sorting here? If we don't collapse, there's no reason to do the sort. I think it would make sense to move the `sortByKey` into `collapse`, since the data being sorted is an invariant of collapse.